src/commands/selectSignImage.ts

```diff
@@ -0,0 +1,142 @@
+import * as vscode from 'vscode';
+import { logInToDockerCli } from './registries/logInToDockerCli';
+import { checkCli } from '../utils/checksCli';
+import { execAsync } from '../utils/execAsync';
+
+const cliName = 'notation';
+const goToNotationButton = 'Download Notation';
+const NotationURL = 'https://notaryproject.dev/docs/installation/cli/';
+const errorMessage = `Notation not found: `;
+
+
+/**
+ * Parsing the string is necessary because the current notation output
+ * is a tabular string, thus not formatted for immediate use. 
+ * Thus the function changes the key list string into an array of objects
+ */
+export async function sortOutput(Key_String_Values: string ){
+    //if the key_string_values is empty
+    if(Key_String_Values == ""){
+        return "No current keys in notation";
+    }else {
+        var dataArray = new Array();
+        var defaultKey = '';
+        var keyChoices = Key_String_Values.split("\n");
+        for (let i = 1; i < keyChoices.length; i++) { 
+            var values = keyChoices[i].trim().split(/ {2,} /); 
+            var itemObject; 
+            var value = values[0];
+            if (value.indexOf('*')> -1){
+                defaultKey = value.replace('* ', '');
+                itemObject = {label: defaultKey, description: "Current default"};
+            } else {
+                itemObject = {label: value};
+            }  
+            dataArray.push(itemObject);
+        }
+        return [dataArray, defaultKey];
+    }
+}
+
+//adds precreated azure key to notation
+async function addNewKey() { 
+
+    const prompt: boolean = vscode.workspace.getConfiguration("securesupplychain").get('promptForKeyVersion', false);
+
+    var key_Name = await vscode.window.showInputBox({prompt: "Input the cert name"});
+        //If the user escapes or enters an empty string in the inputbox, then exit the function
+        if(key_Name == undefined || key_Name == ''){
+            return;
+        }
+
+    if(prompt){
+        var key_ID = await vscode.window.showInputBox({prompt: "Input the Azure Key ID"});
+        if (key_ID == undefined ||  key_ID == '') {
+            return; // Exit the function
+        }
+
+        const progressOptions: vscode.ProgressOptions = {
+            location: vscode.ProgressLocation.Notification,
+            title: vscode.l10n.t('Adding key...'),
+        };
+        await vscode.window.withProgress(progressOptions, async () => {
+                var output = await execAsync(`${cliName} key add --plugin azure-kv --id ${key_ID} ${key_Name}`);
+                if(output.stdout){
+                    vscode.window.showInformationMessage("Key added");
+                }else{ 
+                    vscode.window.showErrorMessage(output.stderr);
+                }            
+        });  
+
+    }else{
+
+        var akv_Name = await vscode.window.showInputBox({prompt: "Input the azure key vault name"});
```

Comment:
your variable and function naming are not consistent throughout. You are using a mixture of camel case, snake case an occasionally even pascal case. While this is not crucial to functionality it does enhance readability and is a minor correction that can be easily done. Linters can also help with that.